decoder/vaapidecoder_h264.cpp

```diff
@@ -1084,8 +1085,10 @@ bool VaapiDecoderH264::DPB::add(const PicturePtr& picture)
         return output(picture);
     }
 
-    while (isFull())
-        bump();
+    while (isFull()) {
+        if (!bump())
```

Comment:
Maybe you misunderstand it. if it is full , we need to bump it. But if we get error from the bump function, we need jump out the loop.